src/System.Web.Http/Batch/DefaultHttpBatchHandler.cs

```diff
@@ -187,7 +187,7 @@ public virtual Task<HttpResponseMessage> CreateResponseMessageAsync(IList<HttpRe
             foreach (HttpContent httpContent in streamProvider.Contents)
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                HttpRequestMessage innerRequest = await httpContent.ReadAsHttpRequestMessageAsync();
+                HttpRequestMessage innerRequest = await httpContent.ReadAsHttpRequestMessageAsync(request.RequestUri.Scheme);
```

Comment:
To avoid changes to unrelated tests, please make this `null`-conditional e.g. `request.RequestUri?.Scheme`. If the repo disallows new-ish C# features, use a ternary expression.

Reply:
Done using a ternary. It's C# 5 so no ?.